alf/metrics/metric.py

```diff
@@ -58,14 +60,18 @@ def result(self):
         raise NotImplementedError(
             'Metrics must define a result() member function')
 
-    def gen_summaries(self, train_step=None, step_metrics=()):
+    def gen_summaries(self,
+                      train_step=None,
+                      step_metrics=(),
+                      other_steps: Dict[str, Number] = dict()):
         """Generates summaries against train_step and all step_metrics.
 
         Args:
             train_step: (Optional) Step counter for training iterations. If None, no
                 metric is generated against the global step.
             step_metrics: (Optional) Iterable of step metrics to generate summaries
                 against.
+            other_steps: A dictionary of steps to generate summaries against.
```

Comment:
It's better to clarify that `other_steps` should directly contain integers instead of metrics, to differentiate from `step_metrics`.

Reply:
The type of other_steps indicates so.